src/signals-mach.c

```diff
@@ -891,3 +892,54 @@ JL_DLLEXPORT void jl_profile_stop_timer(void)
     profile_all_tasks = 0;
     uv_mutex_unlock(&bt_data_prof_lock);
 }
+
+// The mprotect implementation in signals-unix.c does not work on macOS/aarch64, as mentioned.
+// This implementation comes from dotnet, but is similarly dependent on undocumented behavior of the OS.
+// Copyright (c) .NET Foundation and Contributors
+// MIT LICENSE
```

Comment:
We should probably add this to THIRDPARTY.md. I don't know if this is enough code to qualify as copywrite, but we might as well give the acknowledgment credit there since that is all MIT asks for.